baseplate/sidecars/trace_publisher.py

```diff
@@ -176,21 +211,11 @@ def publish_traces() -> None:
         post_timeout=publisher_cfg.post_timeout,
     )
 
-    while True:
-        message: Optional[bytes]
-
-        try:
-            message = trace_queue.get(timeout=0.2)
-        except TimedOutError:
-            message = None
-
-        try:
-            batcher.add(message)
-        except BatchFull:
-            serialized = batcher.serialize()
-            publisher.publish(serialized)
-            batcher.reset()
-            batcher.add(message)
+    if publisher_cfg.queue_type == QueueType.IN_MEMORY.value:
+        with publisher_queue_utils.start_queue_server(host="127.0.0.1", port=9090):
```

Comment:
It might be more elegant to add `start`, `stop`, `__enter__` and `__exit__` methods to the `MessageQueue` class. `__enter__` and `__exit__` could call `start` and `stop` respectively. `PosixMessageQueue` would have an empty handler in both cases, while the `RemoteMessageQueue` would start and stop the server in them. You could then just do

```py
with trace_queue:
    build_and_publish_batch(trace_queue, batcher, publisher, QUEUE_TIMEOUT)
```

Reply:
Would we want the remote queue starting a server? I though RemoteMessageQueue was the client for a remote queue server running in the side-car?

Reply:
Yeah, Tom is right here - the server should be started in a separate process from within the sidecar, I dont think we want the queue itself starting it. Both the sidecar & the main baseplate app use `RemoteMessageQueue` to interact with the server.